lib/flagsmith.rb

```diff
@@ -178,34 +203,65 @@ def get_identity_flags_from_document(identifier, traits = {})
       Flagsmith::Flags::Collection.from_feature_state_models(
         engine.get_identity_feature_states(environment, identity_model),
         analytics_processor: analytics_processor,
-        default_flag_handler: default_flag_handler
+        default_flag_handler: default_flag_handler,
+        offline_handler: offline_handler
```

Comment:
Given the logic below, I'm not too sure why we have to provide the offline handler in to the Flags model class too... but maybe it will become apparent as I continue the review.